packages/schematics/angular/guard/files/__name@dasherize__.guard.ts

```diff
@@ -1,14 +1,24 @@
 import { Injectable } from '@angular/core';
-import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { <%= implementationImports %>ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree } from '@angular/router';
 import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
-export class <%= classify(name) %>Guard implements CanActivate {
-  canActivate(
+export class <%= classify(name) %>Guard implements <%= implementations %> {
+  <% if (implements.includes('CanActivate')) { %>canActivate(
     next: ActivatedRouteSnapshot,
     state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
     return true;
   }
+  <% } %><% if (implements.includes('CanActivateChild')) { %>canActivateChild(
```

Comment:
you can eventually use the enum type here.

Reply:
@alan-agius4 Are you saying to create an enum for the values instead of comparing hard coded strings? If yes would this live in the index.ts file or should it have it’s own file?

Reply:
Hi, he enum is created from the json schema, you don't need to redefine it. 

You can run `yarn build` it will generate an new schema with the enum you created in the json schema.

So you can import it like such 
```ts
import { Schema as GuardOptions, Implement } from './schema';
```

Reply:
Oh very cool. I will make these changes later today when I get home. Thanks!